src/slack_hooks/service/librato_snapshots.clj

```diff
@@ -0,0 +1,244 @@
+(ns slack-hooks.service.librato-snapshots
+  (:require [clojure.data.json :as json]
+            [clojure.string :as str]
+            [slack-hooks.slack :as slack]
+            [clj-time.coerce :as c]
+            [clj-time.core :as t]
+            [clj-http.client :as client]))
+
+(def librato-username
+   (or
+     (System/getenv "LIBRATO_USERNAME")
+     "Librato Alert Snapshots"))
+
+(def librato-avatar
+    (System/getenv "LIBRATO_AVATAR"))
+
+(def librato-slack-url
+   (System/getenv "LIBRATO_SLACK_URL"))
+
+
+;  - If chart with the Alert Name doesn't exist, create it.
+;  - Remove all metrics inside the chart.
+;  - Load metrics that were sent in the webhook. DONE
+;  - Snapshot that Chart
+;  - Send the Slack message
+
+(def librato-api-user
+  (System/getenv "LIBRATO_API_USER"))
+
+(def librato-api-token
+  (System/getenv "LIBRATO_API_TOKEN"))
+
+(def chart-duration
+  (or
+    (System/getenv "LIBRATO_CHART_DURATION")
+    "3600"))
+
+(def librato-space-id
+    (System/getenv "LIBRATO_SPACE_ID"))
+
+(defn librato-api-url
+  [path]
+    (str/join ["https://metrics-api.librato.com" path])
+  )
+
+(def librato-request-options
+  {
+    :basic-auth [librato-api-user librato-api-token]
+    :content-type :json
+    :accept :json
+    :throw-entire-message? true
+  })
+
+(defn snapshot-image
+  "Requests the Snapshot. Tries 5 times to get the image URL. Waits 2 seconds between each try.
+  Returns nil if it can't get the URL."
+   [poll-url times]
+  (let [
+       request (client/get poll-url librato-request-options)
+       image (get (json/read-str (:body request)) "image_href")
+       ]
+       (if (> times 5)
+         nil
+         (if (not (nil? image))
+           image
+           (or (Thread/sleep 2000) (snapshot-image poll-url (+ times 1))
```

Comment:
This is hosted on top of Jetty, so I would imagine this won't block any other slack hook.